internal/store/build_result.go

```diff
@@ -127,8 +128,8 @@ func NewDockerComposeDeployResult(id model.TargetID, containerID container.ID, s
 type K8sBuildResult struct {
 	id model.TargetID
 
-	// The UIDs that we deployed to a Kubernetes cluster.
-	DeployedUIDs []types.UID
+	// DeployedEntities are references to the objects that we deployed to a Kubernetes cluster.
+	DeployedEntities []v1.ObjectReference
```

Comment:
checking my understanding: the main change in this PR (besides moving the UIDSet type to the k8s package) is storing ObjectReferences here instead of UIDs, since the former contains namespace info and we'll need that at some point in the near future?

Reply:
Yes! Right now, the watch logic looks at the "things we parsed out of YAML that we think we'll deploy" to find namespaces to watch for entity changes and then uses "UIDs of things we _actually_ deployed" to decide what entities to actually react to changes thereof.

Since the apiserver model will have an explicit spec, it is kinda weird to have something like this:
```yaml
uids: ["abc123", "def456", "xyz789"]
namespaces: ["mix", "meow"]
```
(There's no relationship between indices of UID + namespace)
vs
```yaml
watch:
  - uid: "abc123"
     namespace: "meow"
  - uid: "def456"
     namespace: "mix"
  - uid: "xyz789"
     namespace: "meow"
```